test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+        self.tmpdir = pathlib.Path(os.getenv('TMPDIR', '/tmp'))
+        self.tmpdir = self.tmpdir / ('scylla-'+self.cfg["host"])
+
+        # Cleanup any remains of the previously running server in this path
+        shutil.rmtree(self.tmpdir, ignore_errors=True)
+        pathlib.Path(self.cfg["workdir"]).unlink(missing_ok=True)
+
+        pathlib.Path(self.tmpdir).mkdir(parents=True, exist_ok=True)
+        pathlib.Path(self.cfg["workdir"]).symlink_to(self.tmpdir)
+        pathlib.Path(self.config_file_name).parent.mkdir(parents=True, exist_ok=True)
+        # Create a configuration file.
+        with open(self.config_file_name, 'w') as config_file:
+            config_file.write(SCYLLA_CONF_TEMPLATE.format(**self.cfg))
+
+        self.log_file = open(self.log_file_name, "wb")
+
+    def find_log_file_pattern(self, fil, pattern_re) -> bool:
+        for line in fil.readlines():
+            if pattern_re.match(line):
+                return True
+        return False
+
+    async def cql_is_up(self) -> bool:
+        """Test that CQL is serving (a check we use at start up)."""
+        caslog = logging.getLogger('cassandra')
+        oldlevel = caslog.getEffectiveLevel()
+        # Be quiet about connection failures.
+        caslog.setLevel('CRITICAL')
+        auth = PlainTextAuthProvider(username='cassandra', password='cassandra')
+        try:
+            with Cluster(contact_points=[self.cfg["host"]], auth_provider=auth) as cluster:
+                with cluster.connect():
+                    return True
+        except NoHostAvailable:
+            return False
+        finally:
+            caslog.setLevel(oldlevel)
+        # Any other exception may indicate a problem, and is passed to the caller.
+
+    async def rest_api_is_up(self) -> bool:
+        """Test that the Scylla REST API is serving. Can be used as a
+        checker function at start up."""
+        try:
+            async with aiohttp.ClientSession() as s:
+                url = "http://{}:10000/".format(self.cfg["host"])
+                async with s.get(url):
+                    return True
+        except aiohttp.ClientConnectionError:
+            return False
+        # Any other exception may indicate a problem, and is passed to the caller.
+
+    async def start(self) -> None:
+        """Start an installed server. May be used for restarts."""
+        START_TIMEOUT = 300     # seconds
+
+        # Add suite-specific command line options
+        scylla_args = SCYLLA_CMDLINE_OPTIONS + self.cmdline_options
+        env = os.environ.copy()
+        env.clear()     # pass empty env to make user user's SCYLLA_HOME has no impact
+        self.cmd = await asyncio.create_subprocess_exec(
+            self.exe,
+            *scylla_args,
+            cwd=self.cfg["workdir"],
+            stderr=self.log_file,
+            stdout=self.log_file,
+            env=env,
+            preexec_fn=os.setsid,
+        )
+
+        self.start_time = time.time()
```

Comment:
Seems like it could be a local variable

Reply:
It might be useful as a non-local one in the future. Overall I don't think it matters.